scipy/special/special/specfun/specfun.h

```diff
@@ -5621,7 +5621,7 @@ void sphj(T x, int n, int *nm, T *sj, T *dj) {
         f1 = 1e-100;
         for (k = m; k >= 0; k--) {
             f = (2.0*k + 3.0)*f1/x - f0;
-            if (k <= *nm) { sj[k - 1] = f; }
+            if (k <= *nm) { sj[k] = f; }
```

Comment:
This is what was causing the segfault. 

https://github.com/scipy/scipy/blob/1d3a067c2ccd0a6efddeb3194163aa9a3879d26e/scipy/special/specfun/specfun.f#L9730-L9732